src/uu/df/src/filesystem.rs

```diff
@@ -68,7 +68,7 @@ where
         path.as_ref().to_path_buf()
     };
     let matches = mounts.iter().filter(|mi| path.starts_with(&mi.mount_dir));
-    matches.max_by_key(|mi| mi.mount_dir.len())
+    matches.min_by_key(|mi| mi.mount_dir.len())
```

Comment:
Can you explain why this fixes the problem? I'm not able to figure it out. A comment would be very helpful. Also, the documentation for the function needs to be updated as well.

Reply:
I can't explain it. I made a thinking error and concluded from the individual case that it will work in other cases too. Which is not the case. I tried it with another file system on my local system and it didn't work... So I guess my current approach to resolve this issue is flawed and I close this PR. Sorry for wasting your time.